gcloud-java-compute/src/main/java/com/google/gcloud/compute/Operation.java

```diff
@@ -0,0 +1,800 @@
+      }
+      if (operationPb.getEndTime() != null) {
+        endTime = TIMESTAMP_FORMATTER.parseMillis(operationPb.getEndTime());
+      }
+      if (operationPb.getError() != null && operationPb.getError().getErrors() != null) {
+        errors =
+            Lists.transform(operationPb.getError().getErrors(), OperationError.FROM_PB_FUNCTION);
+      }
+      if (operationPb.getWarnings() != null) {
+        warnings = Lists.transform(operationPb.getWarnings(), OperationWarning.FROM_PB_FUNCTION);
+      }
+      httpErrorStatusCode = operationPb.getHttpErrorStatusCode();
+      httpErrorMessage = operationPb.getHttpErrorMessage();
+      description = operationPb.getDescription();
+    }
+
+    Builder id(String id) {
+      this.id = id;
+      return this;
+    }
+
+    Builder creationTimestamp(Long creationTimestamp) {
+      this.creationTimestamp = creationTimestamp;
+      return this;
+    }
+
+    Builder operationId(OperationId operationId) {
+      this.operationId = checkNotNull(operationId);
+      return this;
+    }
+
+    Builder clientOperationId(String clientOperationId) {
+      this.clientOperationId = clientOperationId;
+      return this;
+    }
+
+    Builder operationType(String operationType) {
+      this.operationType = operationType;
+      return this;
+    }
+
+    Builder targetLink(String targetLink) {
+      this.targetLink = targetLink;
+      return this;
+    }
+
+    Builder targetId(String targetId) {
+      this.targetId = targetId;
+      return this;
+    }
+
+    Builder status(Status status) {
+      this.status = status;
+      return this;
+    }
+
+    Builder statusMessage(String statusMessage) {
+      this.statusMessage = statusMessage;
+      return this;
+    }
+
+    Builder user(String user) {
+      this.user = user;
+      return this;
+    }
+
+    Builder progress(Integer progress) {
+      this.progress = progress;
+      return this;
+    }
+
+    Builder insertTime(Long insertTime) {
+      this.insertTime = insertTime;
+      return this;
+    }
+
+    Builder startTime(Long startTime) {
+      this.startTime = startTime;
+      return this;
+    }
+
+    Builder endTime(Long endTime) {
+      this.endTime = endTime;
+      return this;
+    }
+
+    Builder errors(List<OperationError> errors) {
+      this.errors = ImmutableList.copyOf(checkNotNull(errors));
+      return this;
+    }
+
+    Builder warnings(List<OperationWarning> warnings) {
+      this.warnings = ImmutableList.copyOf(checkNotNull(warnings));
+      return this;
+    }
+
+    Builder httpErrorStatusCode(Integer httpErrorStatusCode) {
+      this.httpErrorStatusCode = httpErrorStatusCode;
+      return this;
+    }
+
+    Builder httpErrorMessage(String httpErrorMessage) {
+      this.httpErrorMessage = httpErrorMessage;
+      return this;
+    }
+
+    Builder description(String description) {
+      this.description = description;
+      return this;
+    }
+
+    /**
+     * Creates an object.
```

Comment:
Also if this builder is supposed to be package scope, we can probably remove this javadoc and make build() package scope, right?